ebu_tt_live/bindings/__init__.py

```diff
@@ -580,24 +580,29 @@ class LiveStyledElementMixin(StyledElementMixin):
 # removed from version 1.0 of the specification.
 
 class RoleMixin(object):
-    _computed_roles = None
+    _computed_roles = set()
+
     def _semantic_compute_roles(self, dataset):
+        current_roles = set()
         if self.role is not None:
-            self._computed_roles = self.role
-        else:
-            if dataset.get('metadata_roles') is not None and len(dataset['metadata_roles']) > 0:
-                self._computed_roles = dataset['metadata_roles'][-1]
-            else:
-                self._computed_roles = None
+            current_roles.update(self.role)
+
+        if dataset.get('role_stack') and dataset['role_stack']:
```

Comment:
I don't understand what the `and dataset['role_stack']` is for here - there seems to be something missing. Should it be:

```suggestion
        if dataset.get('role_stack') is not None and len(dataset['role_stack'])>0:
```

?

Reply:
When I was working on this project, I am also learning python. :) both ways are correct. I just wanted to simply in a Pythonic way. I had used this approach in Nodejs projects. I just checked if this is working in Python as in JS. And yes it is working :) 


for example this tests are passing for both.

```
    def test_role_stack_conditions(self):
  
        test_cases = [
            ({'role_stack': []}, False),
            ({'role_stack': ['role1', 'role2']}, True),
            ({'role_stack': None}, False),
            ({}, False)
        ]

        for dataset, expected in test_cases:
            with self.subTest(case=dataset):

                # Test the Pythonic way
                pythonic_result = bool(dataset.get('role_stack') and dataset['role_stack'])
                # Test the explicit way
                explicit_result = bool(dataset.get('role_stack') is not None and len(dataset['role_stack']) > 0)

                self.assertEqual(pythonic_result, explicit_result, "Both methods should yield the same result")
                self.assertEqual(pythonic_result, expected, f"Expected {expected} for dataset {dataset}")
```

Reply:
sure, I am updating it right now